clients/privacy-center/cypress/e2e/consent-banner-tcf.cy.ts

```diff
@@ -723,6 +725,87 @@ describe("Fides-js TCF", () => {
         });
       });
 
+      it("skips saving preferences to API when custom API fn is provided in Fides.init", () => {
+        const apiOptions = {
+          /* eslint-disable @typescript-eslint/no-unused-vars */
+          savePreferencesFn: (
+            fides: FidesConfig,
+            consent: CookieKeyConsent,
+            fides_string: string | undefined,
+            experience: PrivacyExperience
+          ) => {},
+          /* eslint-enable @typescript-eslint/no-unused-vars */
+        };
+        cy.spy(apiOptions, "savePreferencesFn").as("mockSavePreferencesFn");
+        cy.fixture("consent/experience_tcf.json").then((privacyExperience) => {
+          stubConfig({
+            options: {
+              isOverlayEnabled: true,
+              tcfEnabled: true,
+              api: apiOptions,
+            },
+            experience: privacyExperience.items[0],
+          });
+        });
+        cy.waitUntilFidesInitialized().then(() => {
+          cy.get("#fides-modal-link").click();
+          cy.getByTestId("consent-modal").within(() => {
+            cy.get("button").contains("Opt out of all").click();
+            // timeout means API call not made, which is expected
+            cy.on("fail", (error) => {
+              if (error.message.indexOf("Timed out retrying") !== 0) {
+                throw error;
+              }
+            });
+            // check that preferences aren't sent to Fides API
+            cy.wait("@patchPrivacyPreference", {
+              requestTimeout: 100,
+            }).then((xhr) => {
+              assert.isNull(xhr?.response?.body);
+            });
+          });
+          cy.wait("@mockSavePreferencesFn");
+          // The cookie should still get updated
+          cy.getCookie(CONSENT_COOKIE_NAME).then((cookie) => {
+            const cookieKeyConsent: FidesCookie = JSON.parse(
+              decodeURIComponent(cookie!.value)
+            );
```

Comment:
These assertions feel like the wrong thing to test here- yes, the cookie should be updated but we don't need to fully interrogate it here, do we?

Instead, can you do some checks that the `savePreferencesFn` works as expected? Some suggestions:
* assert that the `fides_string` passed in matches what we'd expect it to be after clicking "opt out"
* assert that the `experience` passed in matches the one from the stubConfig
* etc